### PR TITLE
incorporating engine version in workgroups

### PR DESCRIPTION
**Pull request by soojinj**
*Description of changes:*
Adding engine version to workgroup, to be reflected in update/create/list wg apis

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: workgroup/pom.xml [hunk 1]
```diff
@@ -30,7 +30,7 @@
         <dependency>
             <groupId>software.amazon.awssdk</groupId>
             <artifactId>athena</artifactId>
-            <version>2.10.40</version>
+            <version>2.16.4</version>
```
Comment (shengnaw) [inline workgroup/pom.xml:33]: Try to make sure this update is good.

File: workgroup/aws-athena-workgroup.json [hunk 2]
```diff
@@ -29,6 +29,26 @@
                 "Value"
             ]
         },
+        "SelectedEngineVersion": {
+            "description": "The engine version requested by the user. Possible values are determined by the output of ListEngineVersions, including Auto. The default is Auto.",
+            "type": "string"
+        },
+        "EffectiveEngineVersion": {
+            "description": "Read only. The engine version on which the query runs. If the user requests a valid engine version other than Auto, the effective engine version is the same as the engine version that the user requested. If the user requests Auto, the effective engine version is chosen by Athena. When a request to update the engine version is made by a CreateWorkGroup or UpdateWorkGroup operation, the EffectiveEngineVersion field is ignored.",
```
Comment (cmihran) [inline workgroup/aws-athena-workgroup.json:37]: Does this need to be marked as ReadOnly in schema? There is a readOnlyProperties property but I'm not sure if that's only for top-level properties. Was curious if you had looked into this at all

File: workgroup/aws-athena-workgroup.json [hunk 3]
```diff
@@ -29,6 +29,26 @@
                 "Value"
             ]
         },
+        "SelectedEngineVersion": {
+            "description": "The engine version requested by the user. Possible values are determined by the output of ListEngineVersions, including Auto. The default is Auto.",
```
Comment (shengnaw) [inline workgroup/aws-athena-workgroup.json:33]: Maybe give an example apart from Auto?

Comment (soojinj) [reply workgroup/aws-athena-workgroup.json:33]: didnt do on purpose as values other than Auto will change time over time. I did list that possible values are determined through listEngineVersion call

File: workgroup/pom.xml (hunk 1 above)
Comment (soojinj) [reply workgroup/pom.xml:33]: it was the latest that had the workgroup + engine version release

File: workgroup/aws-athena-workgroup.json (hunk 2 above)
Comment (soojinj) [reply workgroup/aws-athena-workgroup.json:37]: done

**Review by cmihran — APPROVED** [review]

**Review by shengnaw — APPROVED** [review]

**Pull request merged**